### ISSUE TITLE
Pull Request Merge Order Wish List

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#67]
*edit* updated to add #65  to list

If i had my preference of which PRs should be merged first and in which order:

>Add Travis CI config and fix Junit tests run from Gradle
>#58 opened 3 days ago by DanVanAtta 

- I'm building all new branches based on the branch in this PR. Means you'll see these common commits a lot in recent PRs. Adding travis is really helpful, and fixing the build is just important.

>Game freezes on startup - Game chooser synchronize deadlock fix
>#66 opened 39 minutes ago by DanVanAtta 

- Bug fix! Ever since I did the last merge of the latest reformat changes, my master (and even rolling back to before #58), started seeing this problem. I got a stack dump and found where the deadlock was. Not sure how/why, but the fix is works at least. I think we likely should revisit a lot of the synchronization, I don't think much of it will be correct.

>Remove commented out code from 121 files
>#54 opened 6 days ago by DanVanAtta 

- So many merge conflicts.. Let's get this cruft out of the code base so nobody needs to waste time on it.

> Veqryn final cleanup
> #65 opened 2 hours ago by veqryn  

- my preference is for this to go after the commented code removal. There are lot of merge conflicts between the two, easier to do the automated (reformat) stuff after the by-hand stuff (commented code removal).

**Comment by veqryn** [comment]
i won't add the libraries until after our release, otherwise they make the jar size larger

**Comment by DanVanAtta** [comment]
No problem, can be skipped. The preferred merge order would then would be:
#66 (important fix, removes one "synchronization" keyword to mitigate a deadlock problem I'm seeing)

Then probably add a tag and continue merging:
#58 ,#54, #65 (sorry to make you redo the eclipse formatting, but it'll be easier than redoing #54 which was done by hand)

**Comment by ron-murhammer** [comment]
@DanVanAtta would you prefer if we close pull requests after we add comments that we won't merge in the current state? Or would you prefer us to leave them open and you'll either fix or close them?
Example: #61

**Comment by DanVanAtta** [comment]
Closing this thread since the requests action items have been completed - requested PRs have been merged or otherwise closed or dealt with.

@ron-murhammer IMO PR should be closed generally only by the requester. It is my expectation that when a PR is open the reviewers nor requester will allow "the ball" to be in their court for too long of a period of time. A PR that is completely off-base and that should be re-done, can be closed. When that is done, solid direction and guidance is owed on what could/should be done instead.

It's a pretty strong act to outright reject someones work. It is also moving the goal posts to always insist on perfect work as well. Incremental value and complete features is generally the goal. A PR I could see being rejected because it was only half done and could not be merged into the code base. 

So long answer to your question, let PR requests fix or close their PR in general. Also be sure you lose the notion of a PR as a way to gatekeep as well. Code review does find bugs, it has other benefits too. I the only reason you do PR though is to try and ensure you gatekeep the code from bugs or things that you disagree with but add value, then a person would be missing out on a lot of the benefit of code reviews. 

#61 is an even more interested example IMO since 100% of the conversation has been until now unrelated to the actual change.

 That PR AFAIK does nothing for history being in a save game, it doesn't want to do anything about it, it's a convenience thing so you don't have to click N buttons. It's slow in networked games and it's easy to click faster than the game model updates and wind up with index out of bound errors.
